opentelemetry-otlp/src/exporter/http/logs.rs

```diff
@@ -4,9 +4,124 @@ use http::{header::CONTENT_TYPE, Method};
 use opentelemetry::otel_debug;
 use opentelemetry_sdk::error::{OTelSdkError, OTelSdkResult};
 use opentelemetry_sdk::logs::{LogBatch, LogExporter};
+#[cfg(feature = "http-retry")]
+use std::sync::Arc;
 use std::time;
 
+#[cfg(feature = "http-retry")]
+use super::{classify_http_export_error, HttpExportError, HttpRetryData};
+#[cfg(feature = "http-retry")]
+use opentelemetry_sdk::retry::{retry_with_backoff, RetryPolicy};
+#[cfg(feature = "http-retry")]
+use opentelemetry_sdk::runtime::Tokio;
+
 impl LogExporter for OtlpHttpClient {
+    #[cfg(feature = "http-retry")]
+    async fn export(&self, batch: LogBatch<'_>) -> OTelSdkResult {
+        let policy = RetryPolicy {
+            max_retries: 3,
+            initial_delay_ms: 100,
+            max_delay_ms: 1600,
+            jitter_ms: 100,
+        };
+
+        // Build request body once before retry loop since LogBatch contains borrowed data
+        let (body, content_type, content_encoding) = self
+            .build_logs_export_body(batch)
+            .map_err(OTelSdkError::InternalFailure)?;
+
+        let retry_data = Arc::new(HttpRetryData {
+            body,
+            headers: self.headers.clone(),
+            endpoint: self.collector_endpoint.to_string(),
+        });
+
+        retry_with_backoff(
+            Tokio,
+            policy,
+            classify_http_export_error,
+            "HttpLogsClient.Export",
+            || async {
+                // Get client
+                let client = self
+                    .client
+                    .lock()
+                    .map_err(|e| HttpExportError {
```

Comment:
I think the code would be easier to read if there were two constructors for `HttpExportError`. Something like `HttpExportError::new(500, format!("Mutex lock failed: {e}"))` and `HttpExportError::with_retry_after(...)`. It would cut down on the boiler plate.

Reply:
Good catch - have implemented